tests/test_pyxis_collector.py

```diff
@@ -17,3 +19,35 @@ def test_get_manifest_data():
         manifest = get_manifest_data(manifest_id)
 
     assert manifest == example
+
+
+@pytest.mark.django_db
+def test_get_image_by_nvr(requests_mock):
+    nvr = (
+        "ose-cluster-ovirt-csi-operator-container-v4.12.0-202301042354.p0.gfeb14fb.assembly.stream"
+    )
+    with open("tests/data/pyxis/images_by_nvr.json") as data:
+        requests_mock.get(f"{PYXIS_REST_API_URL}/v1/images/nvr/{nvr}", text=data.read())
+
+    get_image_by_nvr(nvr)
+
+    assert CollectorPyxisImage.objects.count() == 4
+    assert CollectorPyxisImageRepository.objects.count() == 1
+
+    for image in CollectorPyxisImage.objects.all():
+        assert image.repos.count() == 1
+        assert image.nvr == nvr
+        assert image.name_label == "openshift/ose-ovirt-csi-driver-operator"
+        assert image.tags == ["v4.12.0-202301042354.p0.gfeb14fb.assembly.stream"]
+        repo = image.repos.first()
```

Comment:
You can just do .get() and then there's no need to check the .count():
```suggestion
        repo = image.repos.get()
```